play.rb

```diff
@@ -0,0 +1,100 @@
+$LOAD_PATH << '.'
+require 'deck'
+require 'dealer'
+require 'player'
+
+def check_bust(player, dealer)
+  if player.total > 21
+    puts 'PLAYER BUST!! Dealer wins! '\
+    "Player total: #{player.total}, Dealer total: #{dealer.total}"
+    end_game
+  end
+  if dealer.total > 21
+    puts 'DEALER BUST!! Player wins! '\
+    "Player total: #{player.total}, Dealer total: #{dealer.total}"
+    end_game
+  end
+end
+
+def check_win(player, dealer)
+  d = 21 - dealer.total # How close dealer total is to 21
+  pl = 21 - player.total # How close player total is to 21
+  puts 'TIE GAME!' if d == pl
+  puts 'DEALER WINS! '\
+  "Player total: #{player.total}, Dealer total: #{dealer.total}" if d < pl
+  puts 'PLAYER WINS! '\
+  "Player total: #{player.total}, Dealer total: #{dealer.total}" if d > pl
+end
+
+def end_game
+  puts 'Thanks for playing! Seeya next time'
+  exit!
+end
+
+# Game starts here...
```

Comment:
A nice cleanup here would be to make the 'game' logic requirable. The unbound script code here will always execute at load time, regardless of whether it's invoked directly from the command line or required by another file, which could lead to unintended consequences. The simplest workaround for that is to wrap a 'main check' conditional around this code:

```ruby
if __FILE__ == $0
  # game logic goes here...
end
```

to only start the game if invoked from the command line. Some method-based pseudocode might look something like:

```ruby
def play_game
  play_player_hand
  play_dealer_hand
  determine_winner
end

if __FILE__ == $0
  play_game
end
```

Alternatively you could invoke the `play_game` method via rake.